charts/linkerd2/templates/_addons.tpl

```diff
@@ -0,0 +1,25 @@
+{- /*
+  This function does the following functions for add-on fields:
+
+  - when an add-on is enabled, it removes the `partials` and `global` sub-fields which are automatically added
+    by helm to pass configuration to sub-charts, and returns the add-on fields.
+
+  - When the add-on is disabled, only the `enabled` flag is returned as we don't need to store the other configuration
+    when an add-on is disabled.
+*/ -}}
+{{- define "linkerd.addons.sanitize-config" -}}
+{{- if .enabled -}}
+{{- $dupValues := . -}}
+{{- if kindIs "map" $dupValues -}}
```

Comment:
I think it's safe to assume this will always be the case, so you can remove this check. If it's not a map, it's better to have it fail loudly instead of ignoring.

Reply:
makes sense!